src/OneSignal.ts

```diff
@@ -731,11 +732,25 @@ export default class OneSignal {
       throw new InvalidArgumentError('url', InvalidArgumentReason.Malformed);
     if (!ValidatorUtils.isValidUrl(icon, { allowEmpty: true, requireHttps: true }))
       throw new InvalidArgumentError('icon', InvalidArgumentReason.Malformed);
-
-    if (subscription.deviceId) {
-      await OneSignalApi.sendNotification(appConfig.appId, [subscription.deviceId], { en : title }, { en : message },
-                                               url, icon, data, buttons);
+    if (!icon) {
+      // get default icon
+      const icons = await MainHelper.getNotificationIcons();
+      icon = getPlatformNotificationIcon(icons);
     }
+
+    this.context.serviceWorkerManager.getRegistration().then(async (registration) => {
+      if (!registration) {
+        Log.error("Service worker registration not available.");
+        return;
+      }
+
+      const options: NotificationOptions = {
+        body: message,
+        data: data,
+        icon: icon,
+      };
+      registration.showNotification(title, options);
```

Comment:
The OneSignal REST API and the brower's `showNotification()` data format is has a ton of differences. While the SDK's internal welcome notification doesn't use many of the features / parameters customer may be doing so, since this is a public API on the OneSignal class.